generator/src/main/java/com/scottlogic/deg/generator/walker/reductive/FieldSpecNodeVisitor.java

```diff
@@ -0,0 +1,40 @@
+package com.scottlogic.deg.generator.walker.reductive;
+
+import com.scottlogic.deg.generator.Field;
+import com.scottlogic.deg.generator.constraints.atomic.AtomicConstraint;
+import com.scottlogic.deg.generator.decisiontree.ConstraintNode;
+import com.scottlogic.deg.generator.decisiontree.visualisation.BaseVisitor;
+import com.scottlogic.deg.generator.fieldspecs.FieldSpec;
+import com.scottlogic.deg.generator.reducer.ConstraintReducer;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class FieldSpecNodeVisitor extends BaseVisitor {
```

Comment:
Can this type be better named to what it does/achieves?

From what i understand it creates a collection FieldSpec's for the given field where each ConstraintNode contains some atomic constraints for the field.

Maybe `ConstraintNodeFieldSpecExtractionVisitor` or something that otherwise better describes that it does and what it can be used for. This type looks like it visits _nodes_ on a `FieldSpec`?.